Solink.AddIn.Helpers.Test/RestartableAddInGeneratorTest.cs

```diff
@@ -83,7 +83,41 @@ private static void AssertGeneratedCode(string expected, CodeCompileUnit compile
                 provider.GenerateCodeFromCompileUnit(compileUnit, writer, options);
             }
             var actual = sb.ToString();
-            Assert.AreEqual(expected, actual);
+            AssertGeneratedLinesEqual(expected, actual);
+        }
+
+        private static void AssertGeneratedLinesEqual(string expected, string actual)
+        {
+            var lineNumber = 0;
+            using (var er = new StringReader(expected))
+            using (var ar = new StringReader(actual))
+            {
+                var checkLine = new Action(() =>
+                {
+                    lineNumber++;
+                    var innerEl = er.ReadLine();
+                    var innerAl = ar.ReadLine();
+                    // must format separately from Assert.AreEqual in case expected or actual contain curly braces
+                    var message = String.Format("First difference is at line {0}", lineNumber);
+                    Assert.AreEqual(innerEl, innerAl, message);
+                });
+                checkLine();                                                 //-----------------
+                checkLine();                                                 // <auto-generated>
+                checkLine();                                                 //     This code was generated by a tool.
+                lineNumber++; er.ReadLine(); ar.ReadLine(); // don't care    //     Runtime Version:4.0.30319.34209
+                checkLine();                                                 //
+                checkLine();                                                 //     Changes to this file may cause incorrect behavior and will be lost if
+                checkLine();                                                 //     the code is regenerated.
+                checkLine();                                                 // </auto-generated>
+                checkLine();                                                 //-----------------
+                string el = er.ReadLine(), al = ar.ReadLine();

```

Comment:
Another problem with reading the line and not asserting the values before reading the next line.